### ISSUE TITLE
[telemetry] Instrument server-side DataTables sorting and filtering

### ISSUE THREAD

**Issue by phax** [phax/ph-oton#52]
## Goal

Instrument server-side DataTables processing in `ph-oton-datatables`, so that slow tables can be attributed to sorting, filtering or sheer row count.

Server-side DataTables requests are the classic "why is this screen slow" case in a ph-oton application, and they currently produce no timing signal at all beyond the generic AJAX invoker numbers.

## Where

`ph-oton-datatables/src/main/java/com/helger/photon/uictrls/datatables/ajax/AjaxExecutorDataTables.java`

```java
public void handleRequest (@NonNull IRequestWebScopeWithoutResponse aRequestScope,
                           @NonNull PhotonUnifiedResponse aAjaxResponse) throws Exception
```

The interesting cost centres are the two private helpers it calls:

```java
private static void _sort (@NonNull DTSSRequestData aRequestData, ...)
private static ICommonsList <DataTablesServerDataRow> _filter (@NonNull DTSSRequestData aRequestData, ...)
```

The method already declares `throws Exception` (`IAjaxExecutor.handleRequest` does), so `Telemetry.withSpanVoidThrowing (..., Exception)` fits with no signature change.

Related executors that may deserve the same treatment once this lands — decide during implementation, do not expand scope by default:
- `AjaxExecutorDataTablesOnDemand.java`
- `AjaxExecutorDataTablesI18N.java`

## What to emit

New `com.helger.photon.uictrls.datatables.CDataTablesTelemetry`:

| Constant | Value |
|---|---|
| `SPAN_DT_REQUEST` | `photon.datatables.request` |
| `SPAN_DT_SORT` | `photon.datatables.sort` |
| `SPAN_DT_FILTER` | `photon.datatables.filter` |
| `METRIC_DT_DURATION` | `photon.datatables.duration` |
| `METRIC_DT_ROWS_TOTAL` | `photon.datatables.rows.total` |
| `METRIC_DT_ROWS_FILTERED` | `photon.datatables.rows.filtered` |
| `ATTR_DT_SORTED` | `photon.datatables.sorted` |
| `ATTR_DT_FILTERED` | `photon.datatables.filtered` |
| `UNIT_MILLIS` | `ms` |
| `UNIT_ROW` | `{row}` |

- **Span** `photon.datatables.request`, kind `INTERNAL`, wrapping `handleRequest`. It nests inside the `photon.ajax.invoke` span from the AJAX invoker issue.
- **Child spans** `photon.datatables.sort` and `photon.datatables.filter` around `_sort (...)` and `_filter (...)`. These are the two operations that actually scale with row count, and separating them is the whole point — a table that is slow to filter needs a different fix from one that is slow to sort.
- **Histogram** `photon.datatables.duration` in ms, attributes `sorted` + `filtered` (booleans — whether the request actually asked for either).
- **Histograms** `photon.datatables.rows.total` and `photon.datatables.rows.filtered` in `{row}`, no attributes. Row counts as a distribution answer "is this table slow because it is genuinely huge?".

## Cardinality — the trap here

The DataTables request carries **user-entered search terms** (`search[value]`, per-column `columns[i][search][value]`). These must never appear in any telemetry — not as a metric attribute (unbounded) and not as a span attribute (they are user input and routinely contain personal data such as names and e-mail addresses being searched for).

Record only *whether* a search was active (`photon.datatables.filtered` = `true`/`false`), never *what* was searched.

The table's object ID (`CPageParam.PARAM_OBJECT`) is a generated per-render ID, not a stable table identifier — do **not** use it as a metric dimension. If a per-table breakdown is wanted, take the page ID from the surrounding `AbstractWebPage` instrumentation instead.

## POM

`ph-oton-datatables` gets `ph-telemetry` transitively via `ph-oton-uicore` → `ph-oton-core` → `ph-oton-mgrs`, but declare it directly since it is used directly (no `<version>` — the BOM is imported in the root POM):

```xml
<dependency>
  <groupId>com.helger.telemetry</groupId>
  <artifactId>ph-telemetry</artifactId>
</dependency>
```

## Acceptance criteria

- [ ] Sort and filter are separately timed as child spans.
- [ ] No search term appears in any span or metric attribute.
- [ ] The per-render object ID is not used as a metric dimension.
- [ ] `mvn install` passes; `README.md` news bullet added.

---

### Shared conventions for all ph-telemetry work in ph-oton

**Facades.** `com.helger.telemetry.Telemetry` for spans, `com.helger.telemetry.TelemetryMetrics` for instruments. If no `ITelemetryTracerSPI` / `ITelemetryMeterSPI` is registered by the deploying application, everything degrades to cheap no-ops — so emit unconditionally, never guard with an "is telemetry enabled" flag.

**No OpenTelemetry in ph-oton.** Only `com.helger.telemetry:ph-telemetry` is ever added. `ph-telemetry-otel` and the OpenTelemetry SDK are the deploying application's business. The version comes from the `ph-telemetry-parent-pom` BOM already imported in the root `pom.xml`, so the new `<dependency>` needs no `<version>`.

**Reference implementation.** `ph-oton-mgrs/src/main/java/com/helger/photon/mgrs/longrun/` — `CLongRunningJobTelemetry` (names), `LongRunningJobMetrics` (instrument holder), `LongRunningJobTelemetry` (package-private emitter). Follow that layout.

**Constants: one class per module** (decided; `CLongRunningJobTelemetry` stays where it is for backwards compatibility since it is public API since 10.4.0):

| Module | Constants class |
|---|---|
| ph-oton-io | `com.helger.photon.io.CIOTelemetry` |
| ph-oton-app | `com.helger.photon.app.CAppTelemetry` |
| ph-oton-audit | `com.helger.photon.audit.CAuditTelemetry` |
| ph-oton-ajax | `com.helger.photon.ajax.CAjaxTelemetry` |
| ph-oton-api | `com.helger.photon.api.CAPITelemetry` |
| ph-oton-security | `com.helger.photon.security.CSecurityTelemetry` |
| ph-oton-mgrs | `CLongRunningJobTelemetry` (existing) + `com.helger.photon.mgrs.CMgrsTelemetry` (new) |
| ph-oton-core | `com.helger.photon.core.CCoreTelemetry` |
| ph-oton-connect | `com.helger.photon.connect.CConnectTelemetry` |
| ph-oton-uicore | `com.helger.photon.uicore.CUICoreTelemetry` |
| ph-oton-datatables | `com.helger.photon.uictrls.datatables.CDataTablesTelemetry` |

Each is `@Immutable`, `final`, with a private constructor, and groups constants under `// === span names ===`, `// === metric instrument names ===`, `// === attribute keys ===`, `// === metric units ===` comments, exactly as `CLongRunningJobTelemetry` does. Every constant gets a Javadoc comment.

**Naming.** Spans and metrics `photon.<area>.<thing>`, attributes `photon.<area>.<key>`. Units follow the OpenTelemetry convention: `ms` for durations, `{request}` / `{login}` / `{job}` / `{session}` for counts.

**Cardinality is a hard rule.** Metric attributes must be bounded and small. Anything unbounded — user IDs, session IDs, concrete URL paths, execution IDs, IP addresses, exception messages — goes on **spans only**, never on a metric. This mirrors `ATTR_JOB_TYPE` (metric) vs `ATTR_JOB_EXECUTION_ID` (span-only) in `CLongRunningJobTelemetry`.

**Span/thread binding.** A span is bound to the thread that created it (see the warning block at the end of `LongRunningJobTelemetry.onJobEnd`). Prefer `Telemetry.withSpan*` so start and end are guaranteed to happen on the same thread; only fall back to a manual `startSpan` + try/finally when the span genuinely has to outlive the method.

**Existing `StatisticsManager` handlers stay.** Where a site already emits ph-commons statistics, telemetry is added *alongside* them — do not remove or change the statistics handlers. The long-term strategy for the two systems is tracked separately.

**Coding style.** phax styleguide: Hungarian notation, `final` parameters, JSpecify `@NonNull`/`@Nullable`, `_` prefix for private methods, private methods declared before their callers, space before `(` and `<`, inline string concatenation in log calls (no SLF4J `{}` placeholders), Apache 2.0 header.

**Definition of done.** `mvn install` passes; a bullet is added to the `v10.5.0 - work in progress` section of `README.md` under "News and noteworthy".

**Comment by phax** [comment]
Part of #55 — see that issue for the shared conventions, the module-level constants class table and the POM impact overview.

**Comment by phax** [comment]
Implemented for **10.6.0**.

### What was added

| File | Kind |
|---|---|
| `ph-oton-datatables/.../datatables/CDataTablesTelemetry.java` | new, public - span / metric / attribute names |
| `ph-oton-datatables/.../datatables/ajax/DataTablesMetrics.java` | new, public - 3 histograms |
| `ph-oton-datatables/.../datatables/ajax/DataTablesTelemetry.java` | new, package-private emitter |
| `ph-oton-datatables/.../datatables/ajax/AjaxExecutorDataTables.java` | modified |
| `ph-oton-datatables/pom.xml` | `com.helger.telemetry:ph-telemetry` declared directly (no `<version>`, BOM) |

### Emission

* Span `photon.datatables.request` (`INTERNAL`) wrapping `handleRequest (...)` via `Telemetry.<Exception> withSpanVoidThrowing (...)` - the method already declared `throws Exception`, so no signature change. It nests inside `photon.ajax.invoke` from #40.
* Child spans `photon.datatables.sort` and `photon.datatables.filter` around the `_sort (...)` and `_filter (...)` calls in `_handleRequest (...)`. Both are placed at the call site rather than inside the two helpers, so the two spans are symmetric and the helpers stay untouched.
* Histogram `photon.datatables.duration` with `sorted` + `filtered`, recorded from a `finally` so a failed request is timed too.
* Histograms `photon.datatables.rows.total` and `photon.datatables.rows.filtered`, recorded from the `nTotalRecords` / `nTotalDisplayRecords` values the method already computes for the response. No attributes.

`bSorted` comes from `aRequestData.directGetAllOrderColumns ().isNotEmpty ()`, `bFiltered` from `aRequestData.isSearchActive ()`.

### Cardinality - the trap

**No search term reaches any span or metric attribute.** Only the two booleans are recorded - `photon.datatables.filtered` says *whether* a search was active, never *what* was searched. The Javadoc of `ATTR_DT_FILTERED` spells out why: `search[value]` and the per-column equivalents are user input and routinely contain the names and e-mail addresses being searched for, so they are unsafe even as a span attribute.

The per-render object ID (`CPageParam.PARAM_OBJECT`) is not used as a dimension either - it is generated per render and is not a stable table identifier. The same Javadoc points at `photon.webpage.id` from #42 for a per-screen breakdown.

### Scope

`AjaxExecutorDataTablesOnDemand` and `AjaxExecutorDataTablesI18N` were **not** instrumented, as the issue advised ("decide during implementation, do not expand scope by default"). The I18N executor only serves static translation resources, and the on-demand executor deserves its own look at what its cost centres actually are rather than a copy of this instrumentation. Both are still covered by the generic `photon.ajax.invoke` span from #40.

### Verification

Full reactor `mvn install` over all 23 modules: BUILD SUCCESS, no test failures. No unit test was added - driving `AjaxExecutorDataTables` needs a populated `UIStateRegistry` and a rendered table, and the issue's acceptance criteria did not ask for one. `README.md` news bullet added under `v10.6.0 - work in progress`.
